### ISSUE TITLE
remove

### ISSUE THREAD

**Issue by DontUseGithub** [mayeaux/nodetube#41]

**Comment by mayeaux** [comment]
There's currently a flag in `.env.settings` that allows recaptcha to be turned off if desired. I can do the same thing with Google Analytics and I plan to have them both off by default.

**Comment by DontUseGithub** [comment]
There are enough alternatives _(Matomo, Fathom, GoatCounter etc.)_ to google analytics so we could stripe it out completely. I don't see any reasons for keeping this spyware inside the repo, it's just bad PR for this project.

Since we don't yet have a reasonably efficient alternative to recaptcha - we could turn the recaptcha off by default for now.

**Comment by mayeaux** [comment]
I don't necessarily disagree with that, personally I've never heard of those alternatives you've listed, but since it's just a JavaScript widget my choice would be to leave it in there with the flag turned off. A lot of people like Google Analytics so it will be convenient to them, if others don't like it they can just keep it turned off. If more people complain about it I'd be happy to axe it and if others say they've used it that would make me lean towards keeping it, we'll see what the community thinks.

**Comment by DontUseGithub** [comment]
>  personally I've never heard of those alternatives you've listed

they're pretty big
https://github.com/matomo-org/matomo - 13k stars
https://github.com/usefathom/fathom - 6k stars
etc
So there are ethical alternatives and I believe we should not encourage the use of google analytics.

**Comment by mayeaux** [comment]
I tend to agree with you, how are the free tiers of the hosted version of those software? I used to use an open-source analytics project (name escapes me atm) but their hosted versions all cost too much that it was a bit cost prohibitive for me when I was just starting this project.

**Comment by DontUseGithub** [comment]
> I tend to agree with you, how are the free tiers of the hosted version of those software?


- You can self-host it on your pine64 sbc or on a raspberry pi for FREE! 
- Or you could find a cheap $2/month hosting provider and self-host it there.
- Alternatively you can sign up for a framaclick account for FREE https://framaclic.org/# (never tested this myself)

**Comment by mayeaux** [comment]
Yeah honestly that's why I don't mind to have GA in there with a flag, because it should definitely be off by default but if a young startup wants to use GA in a NodeTube instance to be as lean as possible I can't really blame them, can you? Maybe let's code up an implementation of matomo as well and then people can pick which is best for them, I think that would be nice from a developer/instance maintainer's perspective
